packages/block-library/src/site-tagline/edit.js

```diff
@@ -22,11 +23,36 @@ export default function SiteTaglineEdit( { attributes, setAttributes } ) {
 		'site',
 		'description'
 	);
+	const { canUserEdit, readOnlySiteTagLine } = useSelect( ( select ) => {
+		const { canUser, getEntityRecord } = select( coreStore );
+		const siteData = getEntityRecord( 'root', '__unstableBase' );
+		return {
+			canUserEdit: canUser( 'update', 'settings' ),
+			readOnlySiteTagLine: siteData?.description,
+		};
+	}, [] );
 	const blockProps = useBlockProps( {
 		className: classnames( {
 			[ `has-text-align-${ textAlign }` ]: textAlign,
+			'wp-block-site-tagline__placeholder':
+				! canUserEdit && ! readOnlySiteTagLine,
 		} ),
 	} );
+	const siteTaglineContent = canUserEdit ? (
+		<RichText
+			allowedFormats={ [] }
+			onChange={ setSiteTagline }
+			aria-label={ __( 'Site tagline text' ) }
+			placeholder={ __( 'Write site tagline…' ) }
+			tagName="p"
+			value={ siteTagline }
+			{ ...blockProps }
+		/>
+	) : (
+		<p { ...blockProps }>
+			{ readOnlySiteTagLine || __( 'Site Tagline placeholder' ) }
+		</p>
```

Comment:
Makes me wonder if in the long term, we should add `readOnly` bool props to some of our components 🤔

Reply:
Can you expand a bit on this with a 'dummy' example?